### PR TITLE
Tidy up old domain name links

### PR DESCRIPTION
**Pull request by timsneath**
Replace instances of:

- [flutter.io]() with [flutter.dev]()
- [www.dartlang.org]() with [dart.dev]()
- [docs.flutter.io]() with [api.flutter.dev]()

Also fix a broken image link and remove a reference to Flutter being a mobile-only SDK

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,13 +21,13 @@ simulators, and hardware for iOS and Android.
...
-A brief summary of the [getting started guide](https://flutter.io/docs/development/tools/ide):
+A brief summary of the [getting started guide](https://flutter.dev/docs/development/tools/ide):
 
-- install the [Flutter SDK](https://flutter.io/docs/get-started/install)
+- install the [Flutter SDK](https://flutter.dev/docs/get-started/install)
 - run `flutter doctor` from the command line to verify your installation
 - ensure you have a supported [IntelliJ development environment](https://www.jetbrains.com/idea/download), either:
   - IntelliJ 2018.1 or more recent, Community or Ultimate Edition, or
-  - Android Studio 3.2 (note: Android Studio Canary versions are generally _not_ supported)
+  - Android Studio 3.2 or later (note: Android Studio Canary versions are generally _not_ supported)
```
Comment (devoncarew) [inline README.md]: Not for this PR, but we should likely remove explicit version references, and just say something like the latest stable versions.

Comment (timsneath) [reply README.md]: Good idea. Is there a different criteria for development of this feature versus using it?

**Comment by stevemessick** [comment]
The "kokoro-windows" presubmit is still in active development. Please ignore its failure; you're good to merge.

**Comment by timsneath** [comment]
Great! Thanks for the heads-up, Steve!

File: README.md (hunk 1 above)
Comment (devoncarew) [reply README.md]: Both flows just need a reasonably recent version of the IDEs (we have more detailed setup for contributors here: https://github.com/flutter/flutter-intellij/blob/master/CONTRIBUTING.md).

**Pull request merged**